src/gaggiuino.ino

```diff
@@ -720,19 +742,29 @@ static void manualPressureProfile(void) {
 //#############################################################################################
 
 static void brewDetect(void) {
-  if ( brewState() ) {
-    openValve();
-    if (!brewActive) {
-      brewJustStarted();
+  static bool paramsReset = true;
+
+  if ( brewState() && !stopOnWeight()) {
+    if(!paramsReset) {
+      brewParamsReset();
+      paramsReset = true;
     }
+    openValve();
     brewActive = true;
-  } else{
+  } else if ( brewState() && stopOnWeight()) {
```

Comment:
stopOnWeight() is implicitly true here if brewState() is checked again, on the next else branch !brewState() is then implicit, so consider removing them for efficiency

Reply:
Yeah it was there for visibility when i was testing, i'm changing the logic anyway lol